packages/opentelemetry-semantic-conventions/src/trace/messaging.ts

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * messaging attribute names defined by the Opetelemetry Semantic Conventions specification
+ * https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/semantic_conventions/messaging.md
+ */
+export const MessagingAttribute = {
+  /**
+   * A string identifying the messaging system.
+   * example: kafka, rabbitmq, sqs, sns
+   *
+   * @remarks
+   * Required.
+   */
+  MESSAGING_SYSTEM: 'messaging.system',
+
+  /**
+   * The message destination name. This might be equal to the span name but is required nevertheless.
+   * example: MyQueue, MyTopic
+   *
+   * @remarks
+   * Required.
+   */
+  MESSAGING_DESTINATION: 'messaging.destination',
+
+  /**
+   * The kind of message destination
+   * allowed values: queue, topic,
+   *
+   * @remarks
+   * Required only if the message destination is either a queue or topic.
+   */
+  MESSAGING_DESTINATION_KIND: 'messaging.destination.kind',
```

Comment:
according to spec this is wrong

```suggestion
  MESSAGING_DESTINATION_KIND: 'messaging.destination_kind',
```

Reply:
Perhaps we should look into generating these with https://github.com/open-telemetry/build-tools/tree/master/semantic-conventions to prevent this from happening in the future?